osu.Game/Rulesets/Edit/Checks/CheckConcurrentObjects.cs

```diff
@@ -33,9 +38,8 @@ public IEnumerable<Issue> Run(BeatmapVerifierContext context)
                 {
                     var nextHitobject = hitObjects[j];
 
-                    // Accounts for rulesets with hitobjects separated by columns, such as Mania.
-                    // In these cases we only care about concurrent objects within the same column.
-                    if ((hitobject as IHasColumn)?.Column != (nextHitobject as IHasColumn)?.Column)
+                    // Some rulesets impose additional requirements for concurrency, such as Mania only considering hitobjects in the same column.
+                    if (!ConcurrentCondition(hitobject, nextHitobject))
```

Comment:
Code-quality wise this is all sorts of messed up. Why is the method named `ConcurrentCondition()` *without any implementors actually checking concurrency*? Why does `ConcurrentCondition()` exist *alongside* the private `areConcurrent()` here?

Please move the body of `areConcurrent()` to `ConcurrentCondition()`, rename the method to `CheckConcurrentObjects()`, and change the mania code to

```csharp
protected override CheckConcurrentObjects(HitObject first, HitObject second)
    => base.CheckConcurrentObjects(first, second) && (first as IHasColumn)?.Column == (second as IHasColumn)?.Column;
```